tiledb/cf/netcdf_engine/_array_converters.py

```diff
@@ -236,9 +270,21 @@ def inject_dim_creator(
                     "Cannot add a new NetCDF dimension converter to an array that "
                     "already contains NetCDF variable to attribute converters."
                 )
-        self._array_registry.inject_dim_creator(
-            DimCreator(shared_dim, tiles, filters), position
-        )
+        self._array_registry.inject_dim_creator(dim_creator, position)
+
+    @property
+    def max_fragment_shape(self):
+        """Maximum shape of a fragment when copying from NetCDF to TileDB.
+
+        For a dense array, this is the shape of dense fragment. For a sparse array,
+        it is the maximum number of coordinates copied for each dimension.
+        """
+        return tuple(dim_creator.max_fragment_length for dim_creator in self)
+
+    @max_fragment_shape.setter
+    def max_fragment_shape(self, value: Sequence[Optional[int]]):
+        for max_fragment_length, dim_creator in zip(value, self):
```

Comment:
Same comment about the length of `zip` parameters

Reply:
I'll add an error check.